yb-voyager/cmd/importData_test.go

```diff
@@ -846,6 +853,636 @@ CREATE TABLE public.foo (
+				if row.DBType == "source" {
+					exportInserts = int(row.ExportedInserts)
+				}
+				if row.DBType == "target" {
+					importInserts = int(row.ImportedInserts)
+				}
+
+			}
+		}
+		if exportInserts == int(streamingInserts) && exportInserts == importInserts {
+			return true
+		}
+
+	}
+	return false
+}
+
+func validateSequence(t *testing.T, startID int64, endId int64, ybConn *sql.DB, tableName string) {
+	ids := []string{}
+	for i := 16; i <= 25; i++ {
+		ids = append(ids, strconv.Itoa(i))
+	}
+	query := fmt.Sprintf("SELECT id from %s where id IN (%s) ORDER BY id;", tableName, strings.Join(ids, ", "))
+	rows, err := ybConn.Query(query)
+	testutils.FatalIfError(t, err, "failed to read data")
+	var resIds []string
+	for rows.Next() {
+		var id int
+		err = rows.Scan(&id)
+		testutils.FatalIfError(t, err, "error scanning rows")
+		resIds = append(resIds, strconv.Itoa(id))
+	}
+
+	assert.Equal(t, ids, resIds)
+}
+
+// Basic Test for live migration with cutover
+func TestBasicLiveMigrationWithCutover(t *testing.T) {
+	ctx := context.Background()
+
+	// Create a temporary export directory.
+	exportDir = testutils.CreateTempExportDir()
+	defer testutils.RemoveTempExportDir(exportDir)
+
+	createSchemaSQL := `CREATE SCHEMA IF NOT EXISTS test_schema;`
+	createTableSQL := `
+CREATE TABLE test_schema.test_live (
+	id SERIAL PRIMARY KEY,
+	name TEXT,
+	email TEXT,
+	description TEXT
+);`
+	insertDataSQL := `
+INSERT INTO test_schema.test_live (name, email, description)
+SELECT
+	md5(random()::text),                                      -- name
+	md5(random()::text) || '@example.com',                    -- email
+	repeat(md5(random()::text), 10)                           -- description (~320 chars)
+FROM generate_series(1, 10);`
+	dropSchemaSQL := `DROP SCHEMA IF EXISTS test_schema CASCADE;`
+
+	// Start Postgres container for live migration
+	postgresContainer := testcontainers.NewTestContainerForLiveMigration("postgresql", nil)
+	if err := postgresContainer.Start(ctx); err != nil {
+		utils.ErrExit("Failed to start Postgres container: %v", err)
+	}
+
+	// Start YugabyteDB container.
+	yugabytedbContainer := testcontainers.NewTestContainer("yugabytedb", nil)
+	if err := yugabytedbContainer.Start(ctx); err != nil {
+		utils.ErrExit("Failed to start YugabyteDB container: %v", err)
+	}
+	postgresContainer.ExecuteSqls([]string{
+		createSchemaSQL,
+		createTableSQL,
+		insertDataSQL,
+	}...)
+
+	yugabytedbContainer.ExecuteSqls([]string{
+		createSchemaSQL,
+		createTableSQL,
+	}...)
+
+	defer postgresContainer.ExecuteSqls(dropSchemaSQL)
+	defer yugabytedbContainer.ExecuteSqls(dropSchemaSQL)
+
+	_, err := testutils.RunVoyagerCommand(postgresContainer, "export data", []string{
+		"--export-dir", exportDir,
+		"--source-db-schema", "test_schema",
+		"--disable-pb", "true",
+		"--export-type", SNAPSHOT_AND_CHANGES,
+		"--yes",
+	}, nil, true)
+	testutils.FatalIfError(t, err, "Export command failed")
+
+	for {
+		if isExportDataStreamingStarted(t) {
```

Comment:
Why do we need this check in the first place? We already have snapshotPhaseCompleted no?

Reply:
import already handles this, no? It waits until export is done?

Reply:
Yes, import handles it. This is not required; remove it and add a sleep for at least 1 second to ensure export is triggered on export-dir, allowing import also to be triggered after that.